overlord/snapstate/snapstate.go

```diff
@@ -427,6 +428,28 @@ func CheckChangeConflict(st *state.State, snapName string, checkConflictPredicat
 	return nil
 }
 
+func plugConnected(st *state.State, snapName, plugName string) bool {
+	repo := ifacerepo.Get(st)
+	conns, err := repo.Connected(snapName, plugName)
+	return err == nil && len(conns) > 0
+}
+
+// defaultContentPlugProviders takes a snap.Info and returns what
+// default providers there are.
+func defaultContentPlugProviders(st *state.State, info *snap.Info) []string {
+	out := []string{}
+	for _, plug := range info.Plugs {
+		if plug.Interface == "content" && !plugConnected(st, info.Name(), plug.Name) {
```

Comment:
This is incorrect, it should not check if the plug is connected, instead it needs to check if there is a snap on the system  already that provides the needed slot and only download the default-provider if that is not the case.